addons/website_livechat/static/src/js/im_livechat.js

```diff
@@ -46,6 +46,24 @@ LivechatButton.include({
         }
     },
 
+    /**
+     * @private
+     * @param {Object} message
+     * @return {Promise}
+     */
+    _sendMessage: function (message) {
+        return this._super.apply(this, arguments).then(function (){
+            if(message.isFeedback){
+                var cookie = utils.get_cookie('im_livechat_session');
+                if (cookie) {
+                    var channel = JSON.parse(cookie);
+                    session.rpc('/im_livechat/visitor_leave_session', {uuid: channel.uuid});
```

Comment:
Should the callers of `_sendMessage` wait until this is done if they're waiting on the original `_sendMessage` to be done?
I would assume so.

Then you should:
```
var leaveLivechatPromise = session.rpc('/im_livechat/visitor_leave_session', {uuid: channel.uuid});
utils.set_cookie('im_livechat_session', "", -1); // remove cookie
return leaveLivechatPromise
```